### DISCUSSION TITLE
EQ operator on FLOAT

### DISCUSSION THREAD

**Discussion by CosminPerRam** [sequelize/sequelize#15658] [Help & Questions] [answered]
Doing a `findAll` on a table with a `where` that specifies `[Op.eq]: 999.99` on a FLOAT field doesn't yield expected behaviour (and that is to return the entries that have the specific FLOAT field value 999.99).

In the documentation, at [Core Concepts/Model Querying - Basics](https://sequelize.org/docs/v6/core-concepts/model-querying-basics/) there is no mention of doing an `EQ` operator on a floating point field being a problem. Also, of all the examples on that page, there is none that contains a floating point number.

Doing an `GTE` operation returns the expected entries, whereas `LTE` does not.

I know that floating point numbers comparisons are a challenge, but I expected sequelize to manage this problem (for example with requiring an Epsion value for such occasions) or to mention the problem on the querying docs page.

DB: MariaDB 10.7
Sequelize: 6.28.0

**Comment by ephys** [comment, marked as answer]
MariaDB already indicates that the equality operator is unsafe to use with floating points (https://mariadb.com/kb/en/floating-point-accuracy/). We have to draw a line somewhere in our documentation and I personally consider "what can be compared to what" to be the database's concern, not the ORM's.

You can create your own function to compare with a delta like this:

```ts
function floatingPointEquals(value: number, delta: number) {
  return {
    [Op.between]: [value - delta, value + delta];
  };
}

User.findAll({
  score: floatingPointEquals(10, 0.0001),
});
```
